cmd/server/pactasrv/user.go

```diff
@@ -50,3 +53,45 @@ func (s *Server) deleteUser(ctx context.Context, request api.DeleteUserRequestOb
 	// TODO(#12) Implement Authorization
 	return errorNotImplemented("deleteUser")
 }
+
+// Returns the logged in user
+// (GET /user/me)
+func (s *Server) FindUserByMe(ctx context.Context, request api.FindUserByMeRequestObject) (api.FindUserByMeResponseObject, error) {
+	user, err := s.findUserByMe(ctx, request)
+	if err != nil {
+		return errToAPIError(err)
+	}
+	return api.FindUserByMe200JSONResponse(*user), nil
+}
+
+func (s *Server) findUserByMe(ctx context.Context, request api.FindUserByMeRequestObject) (*api.User, error) {
+	token, _, err := jwtauth.FromContext(ctx)
+	if err != nil {
+		// TODO(grady) upgrade this to the new error handling strategy after #12
+		return nil, errorUnauthorized("lookup self", "without authorization token")
+	}
+	if token == nil {
+		return nil, errorUnauthorized("lookup self ", "without authorization token")
+	}
+	email, ok := token.PrivateClaims()["emails"]
+	if !ok {
+		return nil, errorUnauthorized("lookup self", "without email claim in token")
+	}
+	emails, ok := email.([]string)
+	if !ok || len(emails) == 0 {
+		return nil, errorInternal(fmt.Errorf("couldn't parse email claim in token"))
+	}
+	// TODO(#18) Handle Multiple Emails in the Token Claims gracefully
+	if len(emails) > 1 {
+		return nil, errorBadRequest("jwt token", "multiple emails in token")
+	}
+	authnID := token.Subject()
+	if authnID == "" {
+		return nil, fmt.Errorf("couldn't find authn id in jwt")
+	}
+	user, err := s.DB.GetOrCreateUserByAuthn(s.DB.NoTxn(ctx), pacta.AuthnMechanism_EmailAndPass, authnID, emails[0])
```

Comment:
I would not use a `GetOrCreate` endpoint for something like this. `GetOrCreate`-ing a user should happen on a dedicated `login` or similar endpoint, it should be an error if a user tries to do anything else without an account in the DB.

There's a bunch of reasons to do this, but the main one is to have a single location where we create accounts in the DB.

Reply:
The other reasons are something like:

1. We shouldn't have exported methods that nobody is using, it increases the package's surface area unnecessarily
2. Given the way our login system works ('get token, send to server'), the only way of account creation that makes sense is `GetOrCreateUser[...]`, we shouldn't make it tempting to create accounts any other way

Reply:
Wait I might be misunderstanding. When does the token from Azure get sent directly to the server? Aren't we always facilitating that exchange through the Frontend? If so wouldn't just having getOrCreateUser make sense at the FE layer too?

Agreed on the idea of one method at the DB layer. - I'll make `CreateUser` testonly.

Reply:
Copying image from Slack:
![image](https://github.com/RMI-PACTA/app/assets/2402010/c8225dea-e8f5-4bbe-8a24-89fd377b207e)

Reply:
The token that would go to a `PACTA` `/login` endpoint would be the RMI-specific token from step 4, and that would get or create the user